### PR TITLE
Cookstyle fixes, Readme linting fixes, Updating to work on newer dependencies, test fixes

### LINKED ISSUE #22: Fix all checks so they pass
### :speaking_head: Foreword

Thank for taking the time to fill this bug report fully. Without it we may not be able to fix the bug, and the issue may be closed without resolution.

## :ghost: Brief Description

The checks are not passing.

## :pancakes: Cookbook version

Version of the cookbook where you are encountering the issue.

## :woman_cook:  Chef-Infra Version

Version of chef-client in your environment.

## :tophat: Platform details

Operating system distribution and release version. Cloud provider if running in the cloud.

## :police_car: Expected behavior

All of the checks including integration tests pass.

## :heavy_plus_sign: Additional context

Add any other context about the problem here. e.g. related issues or existing pull requests.

### PR DESCRIPTION
**Pull request by axelrtgs**
# Description

Cookstyle fixes, Readme linting fixes

## Issues Resolved

Fixes #22 

## Check List

- [ ] All tests pass. See TESTING.md for details.
- [ ] New functionality includes testing.
- [ ] New functionality has been documented in the README if applicable.

### CODE REVIEW

**Comment by kitchen-porter** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This is a big Pull Request.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: attributes/default.rb [hunk 1]
```diff
@@ -1,15 +1,15 @@
...
 
 default['foreman']['path'] = '/usr/share/foreman'
-default['foreman']['version'] = 'stable'
+default['foreman']['version'] = '2.1.4-1'
 default['foreman']['config_path'] = '/etc/foreman'
 
 default['foreman']['use_repo'] = true
 default['foreman']['repo']['uri'] = 'http://deb.theforeman.org/'
-default['foreman']['repo']['components'] = ['stable']
+component_version = node['foreman']['version'].split('.')[0..1].join('.')
```
Comment (tas50) [inline attributes/default.rb:11]: This will probably not work when the version is overwritten above.

File: recipes/default.rb [hunk 2]
```diff
@@ -2,7 +2,6 @@
 # Cookbook:: foreman
 # Recipe:: default
 #
-include_recipe 'hostname'
 ohai 'reload' do
```
Comment (tas50) [inline recipes/default.rb:5]: If you're not setting the hostname you probably don't need to reload ohai

Comment (axelrtgs) [reply recipes/default.rb:5]: Good point. i didnt think setting the hostname should be part of foreman so ill remove that too.

File: metadata.rb [hunk 3]
```diff
@@ -10,15 +10,11 @@
 
 supports 'ubuntu'
```
Comment (tas50) [inline metadata.rb:11]: make sure to set chef_version here to >= 15.0 now that you're using archive_file

File: attributes/default.rb (hunk 1 above)
Comment (axelrtgs) [reply attributes/default.rb:11]: Lots will probably not work 😝 Given the already start of a table flip, i'm hoping to kinda get it working on the latest versions of cookbooks first then start converting it to custom resources and not worry about that kinda stuff. Step 1) make it work step 2) make it pretty step 3) make it stable.

Appreciate the early feedback though keep it coming. will give me some good things to adjust as i power through this.

File: metadata.rb [hunk 4]
```diff
@@ -10,15 +10,11 @@
...
 
-depends 'apt'
 depends 'apache2'
-depends 'bind'
+depends 'bind', '= 1.3.0'
```
Comment (tas50) [inline metadata.rb:14]: You really want to avoid pins like this. You'll end up with a cookbook that can't be depsolved due to pins in other cookbooks >= is your friend. Let the users control pins aggressively if they need to.

Comment (axelrtgs) [reply metadata.rb:14]: its specific during my rewrite phase because any version > than that exact version removes attributes and then its no longer compatible with this cookbook that depends on them. slowly converting the attributes to be compatible with the custom resource driven newer versions. Apache was a whirlwind to sort out. This cookbook was written for like 5 major versions of every dependency ago when attributes were king.

TLDR; plan is to remove all pins, they are just there to help get the ball rolling.

**Comment by axelrtgs** [comment]
Woops i added the in progress label but realized i should've used a draft PR as the label is easy to miss. keep the great feedback coming i'll address it as i further my rewrite and remove the temporary lets get it working patches. Early feed back is always nice.
